src/sentry/integrations/utils/commit_context.py

```diff
@@ -79,14 +83,22 @@ def find_commit_context_for_event_all_frames(
         extra=extra,
     )
 
-    most_recent_blame = max(file_blames, key=lambda blame: blame.commit.committedDate, default=None)
-    # Only return suspect commits that are less than 2 months old
-    selected_blame = (
-        most_recent_blame
-        if most_recent_blame
-        and is_date_less_than_two_months(most_recent_blame.commit.committedDate)
-        else None
-    )
+    # We want the most recent commit that is within MAX_COMMIT_AGE_DAYS of group_first_seen
+    earliest_valid_date = group_first_seen - timedelta(days=MAX_COMMIT_AGE_DAYS)
+    # Rather than sorting blames by blame.commit.committedDate O(n log n) then looping,
+    # do a single loop O(n) on the unsorted blames, keeping track of the blame.commit.committedDate of the best candidate.
```

Comment:
these comments are confusing. i would remove. why add comments talking about an alternate implementation that doesn't exist?